scripts/release/asset_publish.py

```diff
@@ -403,7 +504,13 @@ def _str2bool(v: str) -> bool:
                 if asset.type == assets.AssetType.COMPONENT:
                     # load component and check if environment exists
                     component = load_component(asset.spec_with_path)
-                    if component.type == "command":
+                    if component.type == assets.ComponentType.PIPELINE.value:
+                        if not validate_and_prepare_pipeline_component(
+                            asset.spec_with_path, final_version, registry_name
+                        ):
+                            failure_list.append(asset)
+                            continue
+                    elif component.type == assets.ComponentType.COMMAND.value:
```

Comment:
One way to handle the default value?
```suggestion
                    elif component.type is None or component.type == assets.ComponentType.COMMAND.value:
```